src/match-version.js

```diff
@@ -1,15 +1,21 @@
 const fs = require("fs");
 const core = require("@actions/core");
 
-module.exports = (gitRef, prefix = "refs/tags/") => {
+module.exports = (gitRef, prefix = "") => {
   const rawPackageJson = fs.readFileSync("package.json", "utf8");
   const packageJson = JSON.parse(rawPackageJson);
 
-  if (!gitRef) {
+  const refsTags = "refs/tags/"
+  if (!gitRef.startsWith(refsTags)) {
```

Comment:
Nice!
This breaks the CI, will need to adjust that

Reply:
@scottmwyant it seems we cannot override `GITHUB_REF` env var in the ci run (https://docs.github.com/en/actions/reference/environment-variables#naming-conventions-for-environment-variables)

This means that the current changes breaks our e2e test which I am not too happy with.
Not sure what the solution to this would be.
I think the easiest would be to omit this check for now, or maybe add some flag that could disable this check?

Reply:
I have created a thin e2e action that wraps the original action but allows overwriting `GITHUB_REF`. I have just merged it to master. Can you please rebase your branch to it? Thanks!

Reply:
I missed the e2e test before 😑. I understand why it broke with the changes I proposed.  Your fix seems like it will work.  I don't want to bog down this PR with further discussion on this topic.  I may submit another PR after I think this through a bit.

Reply:
I think we are good now. E2e is passing.
You just need to re-generate the bundle with `npm run publish` and we can merge this.

Reply:
Hmm.  My last commit should have taken care of that.  When I run `npm run publish` followed by `git status` on my system it says: "nothing to commit, working tree clean" but when I run `git diff --exit-code` like you have in CI, it does return a `1`.  I'll have to dig into it.  Any pointers?

Reply:
what files does it show changed?
have you run `npm i` the dependencies where updated and they get bundled in the dist

Reply:
I have pulled your branch, installed the packages, generated the bundle and the dist files are out of sync

Reply:
Should be fixed now; I neglected to run `npm install` before `npm publish`.

Reply:
It seems now only `index.js.map` is differring. Have you commited that file too after regenerating dist?

Reply:
The problem is with line endings.  See the image of my diff.  I think I just have to force unix style line endings to my patch.

![image](https://user-images.githubusercontent.com/29161635/130776369-649d0b4f-ec40-47ce-8afb-7a18a5af6d5b.png)